server/datastore/mysql/migrations/tables/20230814150442_AddHostScriptResultsTable.go

```diff
@@ -0,0 +1,67 @@
+package tables
+
+import (
+	"database/sql"
+	"fmt"
+)
+
+func init() {
+	MigrationClient.AddMigration(Up_20230814150442, Down_20230814150442)
+}
+
+func Up_20230814150442(tx *sql.Tx) error {
+	_, err := tx.Exec(`
+CREATE TABLE host_script_results (
+    id              INT(10) UNSIGNED NOT NULL AUTO_INCREMENT,
+    host_id         INT(10) UNSIGNED NOT NULL,
+
+    -- execution_id is a unique identifier (e.g. UUID) generated for each
+    -- execution of a script.
+    execution_id    VARCHAR(255) NOT NULL,
+
+    -- in the future, we may have a concept of "saved scripts" and in that case
+    -- the host_script_results may be associated with a script_id instead of
+    -- the actual script contents. If that's the case, it may be best to allow
+    -- this field to be NULL (if a saved script is used) but for now we don't
+    -- support this so I'm making it NOT NULL.
+    script_contents TEXT NOT NULL,
+
+    -- output is the combination of stdout and stderr from the script execution.
+    output          TEXT NOT NULL,
+
+    -- runtime is the execution time of the script in seconds, rounded.
+    runtime         INT(10) UNSIGNED NOT NULL DEFAULT 0,
+
+    -- the exit code of the script execution, large enough to not assume too
+    -- much about the possible range (e.g. https://stackoverflow.com/a/328423/1094941)
+    -- It can be NULL to represent that the script results have not been received
+    -- yet, and -1 if the script executed but was terminated abruptly (e.g. due to
+    -- a signal/timeout, same as how Go reports this: https://pkg.go.dev/os#ProcessState.ExitCode).
+    exit_code       INT(10) NULL,
+
+    created_at      TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP,
+    updated_at      TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
+
+    PRIMARY KEY (id),
```

Comment:
just for my own benefit, if you have time. What's your thinking for using both `id` and `execution_id` instead of making `execution_id` be the primary key?

Reply:
thinking about this, one obvious reason would be to prevent collation issues like this one https://github.com/fleetdm/fleet/issues/12271

Reply:
Generally speaking I usually avoid string PKs as they usually take too much space. The collation issue you mention is a good point too, that I hadn't thought about.